core/scripts/vrfv2/testnet/main.go

```diff
@@ -871,39 +893,120 @@ func main() {
 				uint32(*cbGasLimit),
 				uint32(*numWords),
 				uint16(*requests),
+				decimal.RequireFromString(*subFundingAmountJuels).BigInt(),
 			)
 			helpers.PanicErr(err)
 			fmt.Printf("TX %d: %s\n", i+1, helpers.ExplorerLink(e.ChainID, tx.Hash()))
 			txes = append(txes, tx)
 		}
-		fmt.Println("Total number of requests sent:", (*requests)*(*runs))
+
+		coordinatorAddress, err := consumer.COORDINATOR(nil)
+		helpers.PanicErr(err)
+		coordinator, err := vrf_coordinator_v2.NewVRFCoordinatorV2(coordinatorAddress, e.Ec)
+		helpers.PanicErr(err)
+
+		coordinatorOwnerAddress, err := coordinator.Owner(nil)
+		helpers.PanicErr(err)
+
+		fmt.Println("Actual Coordinator Owner Address:", coordinatorOwnerAddress.String())
+		fmt.Println("Provided VRF Owner Address:", *vrfOwnerAddress)
+		if coordinatorOwnerAddress.String() != *vrfOwnerAddress {
+			panic("Actual Coordinator Owner and provided Coordinator Owner Addresses does not match")
+		}
+
+		//coordinatorOwner, err := vrf_owner.NewVRFOwner(coordinatorOwnerAddress, e.Ec)
+		//helpers.PanicErr(err)
+		//
+		//metricsBefore := getVRFOwnerTestConsumerMetrics(*consumerAddress, e)
+
+		var subId uint64
+		var wg sync.WaitGroup
+
+		wg.Add(1)
+
+		go func() {
+			subCreatedChan := make(chan *vrf_coordinator_v2.VRFCoordinatorV2SubscriptionCreated)
+			subCreatedSubscription, err := coordinator.WatchSubscriptionCreated(nil, subCreatedChan, nil)
+			helpers.PanicErr(err)
+			defer subCreatedSubscription.Unsubscribe()
+			subscriptionCreatedEvent := <-subCreatedChan
+			subId = subscriptionCreatedEvent.SubId
+			fmt.Println("VRF Owner Test Consumer's Sub ID:", subId)
+			defer wg.Done()
+		}()
+
+		wg.Wait()
+
+		totalNumberOfRequests := (*requests) * (*runs)
+		fmt.Println("Total number of requests sent:", totalNumberOfRequests)
 		fmt.Println("fetching receipts for all transactions")
+
+		var receipt *types.Receipt
 		for i, tx := range txes {
-			helpers.ConfirmTXMined(context.Background(), e.Ec, tx, e.ChainID, fmt.Sprintf("load test %d", i+1))
+			receipt = helpers.ConfirmTXMined(context.Background(), e.Ec, tx, e.ChainID, fmt.Sprintf("load test %d", i+1))
+
 		}
+
+		blockNumber := receipt.BlockNumber.Uint64()
+
+		//go func() {
+		//subFundedChan := make(chan *vrf_coordinator_v2.VRFCoordinatorV2SubscriptionFunded)
+		//timeout, cancelFunc := context.WithTimeout(context.Background(), 10*time.Second)
+
+		fmt.Println("subId", subId)
+		subFundedIterator, err := coordinator.FilterSubscriptionFunded(&bind.FilterOpts{End: &blockNumber}, []uint64{subId})
```

Comment:
@jinhoonbang @kidambisrinivas any idea why Filter is not working here? I am using Avalanche testnet

Reply:
it will just hang there